rsmtool/configuration_parser.py

```diff
@@ -897,11 +906,11 @@ def validate_config(cls, config, context='rsmtool'):
             if HAS_RSMEXTRA:
                 default_basename = Path(default_feature_subset_file).name
                 new_config['feature_subset_file'] = default_feature_subset_file
-                logging.warning("You requested feature subsets but did not "
-                                "specify any feature file. "
-                                "The tool will use the default "
-                                "feature file {} available via "
-                                "rsmextra".format(default_basename))
+                cls.logger.warning("You requested feature subsets but did not "
```

Comment:
I'm a little bit concerned here about `cls.logger`. I don't see where this attribute would be created.

So, I cloned `rsmextra`, manually installed `factor_analyzer` and then `pip install -e .`ed from the `rsmextra` clone. Afterwards, I created a config file that had `"feature_subset": "A",` (basically I added it to the example config file in this PR).

Then:

```bash
❯ rsmxval config_rsmxval2.json output3
Output directory: output3
Traceback (most recent call last):
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmdev/bin/rsmxval", line 33, in <module>
    sys.exit(load_entry_point('rsmtool', 'console_scripts', 'rsmxval')())
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/rsmxval.py", line 286, in main
    run_cross_validation(abspath(args.config_file),
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/rsmxval.py", line 114, in run_cross_validation
    configuration = configure('rsmxval', config_file_or_obj_or_dict)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/configuration_parser.py", line 80, in configure
    configuration = parser.parse(context=context)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/configuration_parser.py", line 813, in parse
    return Configuration(configdict,
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/configuration_parser.py", line 157, in __init__
    configdict = ConfigurationParser.validate_config(configdict, context=context)
  File "/Users/mmulholland/Documents/rsmtool_dev/rsmtool/rsmtool/configuration_parser.py", line 909, in validate_config
    cls.logger.warning("You requested feature subsets but did not "
AttributeError: type object 'ConfigurationParser' has no attribute 'logger'
```

Reply:
Good catch! Let me look into this.

Reply:
Fixed!